meta/store.go

```diff
@@ -454,15 +457,28 @@ func (s *Store) close() error {
 	}
 	s.opened = false
 
-	// Notify goroutines of close.
-	close(s.closing)
-	// FIXME(benbjohnson): s.wg.Wait()
+	// Close our exec listener
+	if err := s.ExecListener.Close(); err != nil {
+		s.Logger.Printf("error closing ExecListener %s", err)
+	}
+
+	// Close our RPC listener
+	if err := s.RPCListener.Close(); err != nil {
+		s.Logger.Printf("error closing ExecListener %s", err)
+	}
 
 	if s.raftState != nil {
 		s.raftState.close()
-		s.raftState = nil
 	}
 
+	// Notify goroutines of close.
+	s.mu.Unlock()
```

Comment:
Actually, I think this is where we could do with a comment about why we unlock here.